examples/steel/Selectors.LList2.fst

```diff
@@ -0,0 +1,308 @@
+module Selectors.LList2
+
+open Steel.FractionalPermission
+module Mem = Steel.Memory
+module R = Steel.Reference
+
+// friend Steel.SelEffect
+
+#push-options "--__no_positivity"
+noeq
+type cell (a: Type0) = {
+  tail_fuel: Ghost.erased nat;
+  next: ref (cell a);
+  data: a;
+}
+#pop-options
+
+let next #a (c:cell a) : t a = c.next
+let data #a (c:cell a) : a = c.data
+let mk_cell #a (n: t a) (d:a) = {
+  tail_fuel = Ghost.hide 0;
+  next = n;
+  data = d
+}
+
+let null_llist #a = R.null
+let is_null #a ptr = R.is_null ptr
+
+let v_null_rewrite
+  (a: Type0)
+  (_: t_of vemp)
+: Tot (list a)
+= []
+
+let v_c
+  (n: Ghost.erased nat)
+  (#a: Type0)
+  (r: t a)
+  (c: normal (t_of (vptr r)))
+: Tot prop
+= (Ghost.reveal c.tail_fuel < Ghost.reveal n) == true // to ensure vprop termination
+
+let v_c_dep
+  (n: Ghost.erased nat)
+  (#a: Type0)
+  (r: t a)
+  (nllist: (n': Ghost.erased nat) -> (r: t a  { Ghost.reveal n' << Ghost.reveal n }) -> Pure vprop (requires True) (ensures (fun y -> t_of y == list a)))
```

Comment:
Why use << instead of < for precedes on nats?

Reply:
You are right, finally, using `<` surprisingly works. (I remember the decreases clause of `nllist` adds a `<<` refinement to its last argument when used for recursive calls, and so I tried to match this refinement with the `nllist` arguments to auxiliary functions, but `<` appears to be enough.) Thank you Aymeric!